### PR TITLE
core: deprecate Region.from_op_list

### PR DESCRIPTION
**Pull request by superlopuh**
This is the last deprecation in favour of `Region.__init__` one thing to watch out for is that `Region.get([])` produces a region with a single empty block, and `Region([])` produces a region with no blocks. In the now closed variadic init PR, this meant that the new way to spell `Region.get([])` was `Region(Block())` I wonder if we want to make `Region()` create an empty single block region? Or maybe that would be more confusing? If so, maybe this would be the PR to add it.

I look forward to your feedback!

### CODE REVIEW

File: tests/dialects/test_func.py [hunk 1]
```diff
@@ -19,19 +20,23 @@ def test_func():
     c = Addi.get(a, b)
 
     # Create a region to include a, b, c
-    region = Region.from_operation_list([a, b, c])
+    @Builder.region
```
Comment (superlopuh) [inline tests/dialects/test_func.py:23]: I'm not sure what the previous test actually tested, so I rewrote it a bit. My intuition is that it's safe to delete, but maybe I'm missing something.

**Comment by webmiche** [comment]
The empty region thing is something that will come back to haunt as time and time again. See #727 for example. 
unfortunately, I don't really see a solution for it, if we want to connect with MLIR down to these details, other than fixing it in MLIR as well...

**Review by webmiche — APPROVED** [review]
We should fix the empty region thing, other that that, this is gtg.

**Comment by tobiasgrosser** [comment]
I think for now a Regions should be empty the same way a it is done in MLIR. We feel the sugaring should be added at the level of builders.

**Comment by webmiche** [comment]
The problem is that MLIR is not consistent with its empty regions. Sometimes it has no blocks, sometimes it has a block without ops.

**Pull request merged**